Frameworks/CoreText/CTFontDescriptor.mm

```diff
@@ -15,59 +15,124 @@
+                                                        "CTFontDescriptor",
+                                                        NULL, // init
+                                                        NULL, // copy
+                                                        __CTFontDescriptorDeallocate, // deallocate
+                                                        __CTFontDescriptorEqual,
+                                                        __CTFontDescriptorHash,
+                                                        NULL, //
+                                                        __CTFontDescriptorCopyDescription };
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 CTFontDescriptorRef CTFontDescriptorCreateWithNameAndSize(CFStringRef name, CGFloat size) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    CFStringRef keys[] = { kCTFontNameAttribute, kCTFontSizeAttribute };
+    CFTypeRef values[] = { name, CFAutorelease(CFNumberCreate(kCFAllocatorDefault, kCFNumberCGFloatType, &size)) };
+    CFDictionaryRef attributes = CFDictionaryCreate(kCFAllocatorDefault,
+                                                    reinterpret_cast<const void**>(keys),
+                                                    reinterpret_cast<const void**>(values),
+                                                    2, // number of key-value pairs
+                                                    &kCFTypeDictionaryKeyCallBacks,
+                                                    &kCFTypeDictionaryValueCallBacks);
+    CFAutorelease(attributes);
+    return CTFontDescriptorCreateWithAttributes(attributes);
 }
 
 /**
  @Status Interoperable
  @Notes
 */
 CTFontDescriptorRef CTFontDescriptorCreateWithAttributes(CFDictionaryRef attributes) {
-    return (CTFontDescriptorRef)[(__bridge NSDictionary*)attributes copy];
+    size_t memSize = sizeof(struct __CTFontDescriptor) - sizeof(CFRuntimeBase);
+    CTFontDescriptorRef ret =
+        static_cast<CTFontDescriptorRef>(_CFRuntimeCreateInstance(kCFAllocatorDefault, CTFontDescriptorGetTypeID(), memSize, NULL));
+    struct __CTFontDescriptor* mutableRet = const_cast<struct __CTFontDescriptor*>(ret);
+
+    mutableRet->_dictionary = CFDictionaryCreateCopy(kCFAllocatorDefault, attributes);
+    return ret;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 CTFontDescriptorRef CTFontDescriptorCreateCopyWithAttributes(CTFontDescriptorRef original, CFDictionaryRef attributes) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    // New attributes override old ones
+    CFMutableDictionaryRef newAttributes = CFDictionaryCreateMutableCopy(kCFAllocatorDefault, 0, original->_dictionary);
+
+    CFDictionaryApplyFunction(attributes,
+                              [](const void* key, const void* value, void* context) {
+                                  CFDictionarySetValue(reinterpret_cast<CFMutableDictionaryRef>(context), key, value);
```

Comment:
>context [](start = 96, length = 7)

would prefer it if we didn't smuggle things into context if possible I think? we can pass newAttributes as a lambda capture to get more correct typing instead of falling down to void*